usecases/service_accounts_test.go

```diff
@@ -63,297 +64,329 @@ type saHasPermissionTestCase struct {
 var saHasPermissionTestCases = []saHasPermissionTestCase{
 	// No permissions
 	saHasPermissionTestCase{
+		name:     "No Permissions",
 		regPs:    nil,
 		test:     "Service1::RL::Do1::x::*",
 		expected: false,
 	},
 	// != Service
 	saHasPermissionTestCase{
+		name:     "Different Service",
 		regPs:    []string{"Service1::RL::Do1::x::*"},
 		test:     "Service2::RL::Do1::x::*",
 		expected: false,
 	},
 	// Toying around with actions, * and multiple layers in RH
 	saHasPermissionTestCase{
+		name:     "Wildcard operator Scenario 1",
 		regPs:    []string{"Service1::RL::Do1::x::*"},
 		test:     "Service1::RL::Do1::x::*",
 		expected: true,
 	},
 	saHasPermissionTestCase{
+		name:     "Wildcard operator Scenario 2",
 		regPs:    []string{"Service1::RL::Do2::x::*"},
 		test:     "Service1::RL::Do1::x::*",
 		expected: false,
 	},
 	saHasPermissionTestCase{
+		name:     "Wildcard operator Scenario 3",
 		regPs:    []string{"Service1::RL::Do2::x::*", "Service1::RL::Do1::x::*"},
 		test:     "Service1::RL::Do1::x::*",
 		expected: true,
 	},
 	saHasPermissionTestCase{
+		name:     "Wildcard operator Scenario 4",
 		regPs:    []string{"Service1::RL::Do2::x::*", "Service1::RL::Do1::*"},
 		test:     "Service1::RL::Do1::x::*",
 		expected: true,
 	},
 	saHasPermissionTestCase{
+		name:     "Wildcard operator Scenario 5",
 		regPs:    []string{"Service1::RL::*::x::*"},
 		test:     "Service1::RL::Do1::x::*",
 		expected: true,
 	},
 	saHasPermissionTestCase{
+		name:     "Wildcard operator Scenario 6",
 		regPs:    []string{"Service1::RL::*::*"},
 		test:     "Service1::RL::Do1::x::*",
 		expected: true,
 	},
 	saHasPermissionTestCase{
+		name:     "Wildcard operator Scenario 7",
 		regPs:    []string{"Service1::RL::*::*"},
 		test:     "Service1::RL::Do2::*",
 		expected: true,
 	},
 	saHasPermissionTestCase{
+		name:     "Wildcard operator Scenario 8",
 		regPs:    []string{"Service1::RL::Do1::*"},
 		test:     "Service1::RL::Do2::*",
 		expected: false,
 	},
 	// Ownership levels
 	saHasPermissionTestCase{
+		name:     "Different Ownership Level Scenario 1",
 		regPs:    []string{"Service1::RL::Do1::x::*"},
 		test:     "Service1::RO::Do1::x::*",
 		expected: false,
 	},
 	saHasPermissionTestCase{
+		name:     "Different Ownership Level Scenario 2",
 		regPs:    []string{"Service1::RO::Do1::*"},
 		test:     "Service1::RL::Do1::x::*",
 		expected: true,
 	},
 	saHasPermissionTestCase{
+		name:     "Different Ownership Level Scenario 3",
 		regPs:    []string{"Service1::RO::Do1::x::*"},
 		test:     "Service1::RL::Do1::x::*",
 		expected: true,
 	},
 	saHasPermissionTestCase{
+		name:     "Different Ownership Level Scenario 4",
 		regPs:    []string{"Service1::RO::Do1::y::*"},
 		test:     "Service1::RL::Do1::x::*",
 		expected: false,
 	},
 }
 
 func TestServiceAccountsHasPermissionWhenPermissionsOnBaseRole(t *testing.T) {
```

Comment:
Suggestion: I'd name this `TestHasPermissionString_baseRole` (and the following `TestHasPermissionString_nestedRole`)